api/payout.go

```diff
@@ -3,82 +3,160 @@ package api
-func (c *Call) InitiateDirectBulkPayout(ctx context.Context, request model.InitiatePayoutRequest) (model.PayoutDetails, error) {
-	endpoint := fmt.Sprintf("%s/payouts", c.baseURL)
-
-	fL := c.logger.With().Str("func", "InitiateDirectBulkPayout").Str("endpoint", endpoint).Logger()
-	fL.Info().Msg("starting...")
-	fL.Info().Interface(model.LogStrRequest, request).Msg("request")
-	defer fL.Info().Msg("done...")
-
-	errorRes := model.ErrorResponse{}
-	response := struct {
-		Data model.PayoutDetails `json:"data"`
-	}{}
-
-	res, err := c.client.R().
-		SetHeader("request_id", helpers.GetRequestID(ctx)).
-		SetAuthToken(c.bearerToken).
-		SetContext(ctx).
-		SetBody(request).
-		SetResult(&response).
-		SetError(&errorRes).
-		Post(endpoint)
-
-	if err != nil {
-		log.Err(err).Msg("something went wrong with this request")
-		return model.PayoutDetails{}, err
+func (c *Call) InitiateDirectBulkPayout(ctx context.Context, request model.InitiateBulkPayoutRequest) (model.PayoutDetails, error) {
+	var (
+		err      error
+		response model.PayoutDetails
+		path     = payoutAPIVersion
+	)
+
+	err = c.makeRequest(ctx, path, http.MethodPost, nil, nil, nil, request, &response)
+
+	return response, err
+}
+
+// InitiatePayout makes a request to Torus to initiate a bulk payout
+func (c *Call) InitiatePayout(ctx context.Context, currency, payoutType, beneficiaryType, remarks string, document *os.File) (model.PayoutDetails, error) {
+	var (
+		err      error
+		response model.PayoutDetails
+		formData = make(map[string]interface{})
+		path     = fmt.Sprintf("%s/upload", payoutAPIVersion)
+	)
+
+	formData["currency"] = currency
+	formData["payout_type"] = payoutType
+	formData["beneficiary_type"] = beneficiaryType
+	formData["remarks"] = remarks
+	formData["document"] = document
+
+	err = c.makeRequest(ctx, path, http.MethodPost, nil, nil, formData, nil, &response)
+
+	return response, err
+}
+
+// GetAllPayouts makes request to Torus to get all payouts
+func (c *Call) GetAllPayouts(ctx context.Context, status, search string, dateBetween model.DateBetween, page model.Page) (model.AllPayoutsResponse, error) {
+	var (
+		err      error
+		response model.AllPayoutsResponse
+		params   = make(map[string]interface{})
+		path     = payoutAPIVersion
+	)
+
+	if status != "" {
+		params["status"] = status
+	}
+	if search != "" {
+		params["search"] = search
+	}
+	if dateBetween != (model.DateBetween{}) {
+		if dateBetween.From != "" {
+			params["from"] = dateBetween.From
+		}
+		if dateBetween.To != "" {
+			params["to"] = dateBetween.To
+		}
 	}
+	if page != (model.Page{}) {
+		if page.Number != nil {
+			params["number"] = strconv.Itoa(*page.Number)
+		}
+		if page.Size != nil {
+			params["size"] = strconv.Itoa(*page.Size)
+		}
+		if page.SortBy != "" {
+			params["sort_by"] = page.SortBy
+		}
+		if page.SortDirectionDesc != nil {
+			params["sort_direction_desc"] = strconv.FormatBool(*page.SortDirectionDesc)
+		}
+	}
```

Comment:
create an helper function for this. the `dateBetween` also.

Reply:
Alright sir